src/syntaxtree.c

```diff
@@ -273,6 +441,51 @@ static Obj SyntaxTreeIf(Obj result, Stat stat)
     return result;
 }
 
+static Expr SyntaxTreeCodeIf(Obj node)
+{
+    Obj  branches = ElmPRec(node, RNamName("branches"));
+    UInt len = LEN_PLIST(branches);
+    UInt tnum =
+        FIND_TNUM_FROM_STRING(CSTR_STRING(ElmPRec(node, RNamName("type"))));
+    Expr ifexpr = NewStatWithProf(tnum, 2 * len * sizeof(Expr), 0);
+    for (int i = 0; i < len; i++) {
+        Obj  condbodypair = ELM_PLIST(branches, i + 1);
+        Obj  conditionnode = ElmPRec(condbodypair, RNamName("condition"));
+        Obj  bodynode = ElmPRec(condbodypair, RNamName("body"));
+        Expr condition = SyntaxTreeDefaultCoder(conditionnode);
+        Expr body = SyntaxTreeDefaultCoder(bodynode);
+        WRITE_EXPR(ifexpr, 2 * i, condition);
+        WRITE_EXPR(ifexpr, 2 * i + 1, body);
+    }
+    return ifexpr;
+}
+
+static Expr SyntaxTreeCodeImmediateInteger(Obj node)
+{
+    Obj value = ElmPRec(node, RNamName("value"));
+    return INTEXPR_INT(INT_INTOBJ(value));
```

Comment:
We need a check here that verifies `value` really is an immediate integer.

Reply:
We currently do not check the input value in many places. I will add the check, but there are probably even more to add.

Reply:
I will use `Int_ObjInt`, which performs error checking

Reply:
Although this leaves a bit of range here, as it might convert a slightly larger one than the immediate integer. So maybe perform the check by hand?